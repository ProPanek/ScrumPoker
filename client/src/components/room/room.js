import React, {Component} from "react";
import Card from "@material-ui/core/Card";
import Grid from "@material-ui/core/Grid";
import styled from "styled-components";
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import JoinDialog from "../join_room/join_dialog";
import Issue from "./issue/issue";
import CardResults from "./card_results/card_results";
import Controls from "./controls/controls";
import Lists from "./lists/lists";
import {
  BrowserView,
  MobileView
} from "react-device-detect";
import {decorate, observable, when} from "mobx";
import Redirect from "react-router-dom/Redirect";
import routes from "routes"
import ItemPreview from "drag_layer";

const StyledGrid = styled(Grid)`
  &&{
  height: calc(100vh - 48px);
  text-align:center;
  flex:1;
  }
`;

const StyledCard = styled(Card)`
  &&{
  height: calc(100vh - 48px);
  overflow-y: auto;
  text-align:center;
  width:100%;
  }
`;

class Room extends Component {
  componentDidMount() {
    const {store: {userStore: {kicked, setKicked, setConnected}}} = this.props
    this.notFound = window.__ROOM_NOT_FOUND__
    window.onpopstate = this.onBackButtonEvent

    when(
      () => kicked,
      () => {
        this.props.history.push(routes.root());
        setKicked(false);
        setConnected(false)
      },
    )

  }

  onBackButtonEvent = (e) => {
    e.preventDefault()
    const {store: {userStore: {leaveRoom}, socketStore: {openNotification}}, history: {push}} = this.props
    leaveRoom()
    openNotification("You have left the Room", "warning")
    push(routes.root())
  };

  render() {
    return (
      <React.Fragment>
        <ItemPreview/>
        {this.notFound && <Redirect to="/error"/>}
        <BrowserView style={{display: "flex", flexWrap: "wrap", margin: "0 auto", width: "100%"}}>
          <JoinDialog/>
          <StyledGrid item md={10}>
            <StyledCard>
              <Issue/>
              <CardResults/>
              <Controls/>
            </StyledCard>
          </StyledGrid>
          <StyledGrid item md={2}>
            <StyledCard>
              <Lists/>
            </StyledCard>
          </StyledGrid>
        </BrowserView>
        <MobileView style={{display: "flex", flexWrap: "wrap", margin: "0 auto", width: "100%"}}>
          <JoinDialog/>
          <StyledCard>
            <Issue/>
            <CardResults/>
            <Controls/>
          </StyledCard>
          <StyledCard>
            <Lists/>
          </StyledCard>
        </MobileView>
      </React.Fragment>
    );
  }
}

decorate(Room, {
  notFound: observable,
});

export {Room}
export default inject("store")(withRouter(observer(Room)));
