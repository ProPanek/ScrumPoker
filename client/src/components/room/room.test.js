import React from "react"
import {Room} from "./room"
import { shallow } from "enzyme";

const noop = () => {}

describe("Room", () => {
  it("renders without errors", () => {
    const store = {
      roomStore: { cardsAreTheSame : false, cardResults : [], cardHistory: []},
      userStore: {admin: true, users: []},
      jiraStore: {activeBoardFetching: false,  activeBoard: { issues: []}},
      fetchUsers: noop
    }

    const component = shallow(<Room store={store} />)
    expect(component).toMatchSnapshot()
  })
})
