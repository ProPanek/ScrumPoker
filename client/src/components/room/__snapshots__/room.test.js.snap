// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room renders without errors 1`] = `
<Fragment>
  <DragLayer(ItemPreview) />
  <BrowserView
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "margin": "0 auto",
        "width": "100%",
      }
    }
  >
    <inject-withRouter(JoinDialog)-with-store />
    <Styled(WithStyles(Grid))
      item={true}
      md={10}
    >
      <Styled(WithStyles(Card))>
        <inject-Issue-with-store />
        <inject-CardResults-with-store />
        <inject-Controls-with-store />
      </Styled(WithStyles(Card))>
    </Styled(WithStyles(Grid))>
    <Styled(WithStyles(Grid))
      item={true}
      md={2}
    >
      <Styled(WithStyles(Card))>
        <inject-withRouter(Lists)-with-store />
      </Styled(WithStyles(Card))>
    </Styled(WithStyles(Grid))>
  </BrowserView>
  <MobileView
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "margin": "0 auto",
        "width": "100%",
      }
    }
  >
    <inject-withRouter(JoinDialog)-with-store />
    <Styled(WithStyles(Card))>
      <inject-Issue-with-store />
      <inject-CardResults-with-store />
      <inject-Controls-with-store />
    </Styled(WithStyles(Card))>
    <Styled(WithStyles(Card))>
      <inject-withRouter(Lists)-with-store />
    </Styled(WithStyles(Card))>
  </MobileView>
</Fragment>
`;
